Source/Microsoft.Teams.Apps.FAQPlusPlus.Common/Providers/QnaServiceProvider.cs

```diff
@@ -183,19 +183,29 @@ public async Task DeleteQnaAsync(int questionId)
         /// </summary>
         /// <param name="question">Question text.</param>
         /// <param name="isTestKnowledgeBase">Prod or test.</param>
-        /// <returns>QnaSearchResult result as response.</returns>
-        public async Task<QnASearchResultList> GenerateAnswerAsync(string question, bool isTestKnowledgeBase)
+        /// <param name="payload">The response card payload which contains the previous question when querying follow up prompts.</param>
+        /// <returns>QnaSearchResultList result as response.</returns>
+        public async Task<QnASearchResultList> GenerateAnswerAsync(string question, bool isTestKnowledgeBase, ResponseCardPayload payload = null)
         {
             var knowledgeBaseId = await this.configurationProvider.GetSavedEntityDetailAsync(ConfigurationEntityTypes.KnowledgeBaseId).ConfigureAwait(false);
 
-            QnASearchResultList qnaSearchResult = await this.qnaMakerRuntimeClient.Runtime.GenerateAnswerAsync(knowledgeBaseId, new QueryDTO()
+            QueryDTO queryDTO = new QueryDTO()
             {
                 IsTest = isTestKnowledgeBase,
                 Question = question?.Trim(),
                 ScoreThreshold = Convert.ToDouble(this.options.ScoreThreshold),
-            }).ConfigureAwait(false);
+            };
+
+            if (payload != null && payload.IsPrompt)
+            {
+                queryDTO.Context = new QueryDTOContext
+                {
+                    PreviousQnaId = payload.PreviousQuestions.FirstOrDefault().Id.ToString(),
```

Comment:
If the list is empty, FirstOrDefault() will return null, which leads to a null reference exception.
If we're sure that the list will not be empty (i.e., it's an invariant that if IsPrompt=false, PreviousQuestions is not empty), then let's use First().

Reply:
Changed to First() as we are already doing the null check.